account-decoder/src/parse_bpf_loader.rs

```diff
@@ -17,11 +17,11 @@ pub fn parse_bpf_upgradeable_loader(
         UpgradeableLoaderState::Uninitialized => BpfUpgradeableLoaderAccountType::Uninitialized,
         UpgradeableLoaderState::Buffer { authority_address } => {
             let offset = if authority_address.is_some() {
-                UpgradeableLoaderState::buffer_data_offset().unwrap()
```

Comment:
somehow if possible, keeping the `offset` terminology would probably be more readable at call-sites.
(i.e. not immediately clear why the caller should care about `metadata` or size of structs here)

Reply:
Yeah, you're definitely right. I think I would like to abstract this all away from the decoder in a follow up PR.